tests/host_tools/fcmetrics.py

```diff
@@ -363,3 +367,126 @@ def validate_per_device_metrics(self, fc_metrics):
+    return "Count"
+
+
+def flush_fc_metrics_to_cw(fc_metrics, metrics, assert_on_err=False):
+    """
+    Flush Firecracker metrics to CloudWatch
+    Use a new metrics logger and existing dimensions
+    Add a prefix "fc_metrics." to differentiate these metrics.
+    """
+    # List of SharedStoreMetric that once updated have the same value thoughout the life of vm
+    metrics_to_export_once = {
+        "api_server.process_startup_time_us",
+        "api_server.process_startup_time_cpu_us",
+        "latencies_us.full_create_snapshot",
+        "latencies_us.diff_create_snapshot",
+        "latencies_us.load_snapshot",
+        "latencies_us.pause_vm",
+        "latencies_us.resume_vm",
+        "latencies_us.vmm_full_create_snapshot",
+        "latencies_us.vmm_diff_create_snapshot",
+        "latencies_us.vmm_load_snapshot",
+        "latencies_us.vmm_pause_vm",
+        "latencies_us.vmm_resume_vm",
+    }
+    skip = set()
+    for group, keys in fc_metrics.items():
+        if group == "utc_timestamp_ms":
+            continue
+        for key, value in keys.items():
+            full_key = f"{group}.{key}"
+            # values are 0 when:
+            # - there is no update
+            # - device is not used
+            # - SharedIncMetric reset to 0 on flush so if
+            #   there is no change metric the values remain 0.
+            # We can save the amount of bytes we export to
+            # CloudWatch in these cases.
+            # The other cases of 0 are failures in which case
+            # we should assert instead of sending to CW.
+            if (
+                "fail" in full_key
+                or "err" in full_key
+                or "num_faults" in full_key
+                or "panic" in full_key
+            ) and assert_on_err is True:
+                assert 0 == value, f"Error reported in metrics: {full_key}={value}"
+
+            if value != 0:
+                if full_key in skip:
+                    continue
+                unit = get_emf_unit_for_fc_metrics(key)
+                metrics.put_metric(f"fc_metrics.{full_key}", value, unit=unit)
+                # print(f"fc_metrics.{full_key} {value}, unit={unit}")
+                if full_key in metrics_to_export_once:
+                    skip.add(full_key)
+
+
+class FCMetricsMonitor(Thread):
+    """
+    read Firecracker metrics from the microvm every 60secs and
+    uploads the metrics to CW.
+    We do this as a daemon thread every 60sec, instead of collecting
+    all metrics together in the end, to retain timestamp of the metrics.
+    """
+
+    def __init__(self, vm, dimensions, assert_on_err=False):
+        Thread.__init__(self, daemon=True)
+        self.vm = vm
+        self.dimensions = dimensions
+        self.assert_on_err = assert_on_err
+
+        self.metrics_index = 0
+        self.running = True
+
+        self.metrics_logger = get_metrics_logger()
+        self.metrics_logger.set_dimensions(dimensions)
```

Comment:
I'm thinking the metrics_logger could just get passed to this class, so it would also avoid having to deal with dimensions. Since it looks like in the rest of the tests we just use the same dimensions, this would be simpler.